update.sh

```diff
@@ -85,15 +85,53 @@ for ext in "${exts[@]}"; do
 	fi
 
 	for algorithm in md5 sha1 sha256 sha512; do
-		${algorithm}sum "$archive" >> "$cwd/$ruby/checksums.$algorithm"
+		# 1) Append the new checksum line
+		"${algorithm}sum" "$archive" >> "$cwd/$ruby/checksums.$algorithm"
+
+		# 2) Remove duplicates (keeping the first occurrence) without sorting
+		awk '!seen[$0]++' "$cwd/$ruby/checksums.$algorithm" \
+			> "$cwd/$ruby/checksums.$algorithm.tmp"
+
+		mv "$cwd/$ruby/checksums.$algorithm.tmp" "$cwd/$ruby/checksums.$algorithm"
 	done
 	popd >/dev/null
 done
 
+# This script appends a new version to versions.txt and ensures it remains sorted.
+# Steps:
+# 1. Append the new version to versions.txt.
+# 2. Temporarily transform stable versions (those with no dash, e.g., "3.0.0")
+#    by appending "-zzzzzz". This ensures they sort *after* any lines containing
+#    suffixes like "-preview", "-rc", or "-pXYZ".
+# 3. Sort the file uniquely (-u) with "-" as the field separator (-t-):
+#    -k1,1V sorts the main version (e.g., "3.0.0") as a version,
+#    -k2,2V sorts suffixes (e.g., "preview1", "rc2") as versions,
+#    so the transformed stable lines appear last in their version group.
+# 4. Remove the temporary "-zzzzzz" suffix from stable versions.
+# 5. Replace the original file with the sorted result.
 echo "$version" >> "$ruby/versions.txt"
+sed 's/^\([0-9][^-]*\)$/\1-zzzzzz/' "$ruby/versions.txt" \
+| sort -u -t- -k1,1V -k2,2V \
+| sed 's/-zzzzzz$//' \
+> "$ruby/versions.txt.tmp"
+mv "$ruby/versions.txt.tmp" "$ruby/versions.txt"
 
-if [[ $(wc -l < "$ruby/stable.txt") == "1" ]]; then
-	echo "$version" > "$ruby/stable.txt"
+if [[ -f "$ruby/stable.txt" ]]; then
+	stable_file="$ruby/stable.txt"
+	version_family="${version%.*}" # Extract major.minor from version (e.g., 3.3)
+
+	# Use sed to replace the version for the major.minor family or append it if not found
+	if grep -qE "^${version_family}\." "$stable_file"; then
+		sed -i '' -E "s/^(${version_family}\.).*/$version/" "$stable_file"
```

Comment:
* First time I've seen `-i ''` instead of just `-i`.
* `-E` might not be necessary here?
* The `.` in the `version_family` will be interpreted as a regex any-character, instead of `\.`.

Reply:
> First time I've seen `-i ''` instead of just `-i`

When I first ran it with just `-i` it seemed to be trying to use the `-E part in its backup filename. Not sure if I was doing something else wrong, or if this is an intricacy with the macOS flavour of `sort` or similar.

> `-E` might not be necessary here?

Valid.. can definitely test without and remove if not needed.

> The `.` in the `version_family` will be interpreted as a regex any-character, instead of `\.`.

Makes sense 👌🏻

Reply:
With `sed -i -E "s/^(${version_family}\.).*/$version/" "$stable_file"`, it creates the file: `ruby/stable.txt-E`

Reply:
`sed -i '' "s/^(${version_family}\.).*/$version/" "$stable_file"` seems to work fine without the `-E`

`gsed -i "s/^(${version_family}\.).*/$version/" "$stable_file"` also seems to work fine without the `-E`

Reply:
This is because `-i` takes an optional inline suffix.

```
       -i[SUFFIX], --in-place[=SUFFIX]

              edit files in place (makes backup if SUFFIX supplied)
```

Maybe try `-E -i` instead, or try without `-E`.